projects/packages/my-jetpack/src/products/class-creator.php

```diff
@@ -0,0 +1,389 @@
+		return array(
+			'tiers' => array(
+				self::FREE_TIER_SLUG     => array(
+					'available' => true,
+					'is_free'   => true,
+				),
+				self::UPGRADED_TIER_SLUG => array_merge(
+					array(
+						'available'          => true,
+						'wpcom_product_slug' => self::UPGRADED_TIER_PRODUCT_SLUG,
+					),
+					Wpcom_Products::get_product_pricing( self::UPGRADED_TIER_PRODUCT_SLUG )
+				),
+			),
+		);
+	}
+
+	/**
+	 * Get the URL where the user manages the product
+	 *
+	 * @return ?string
+	 */
+	public static function get_manage_url() {
+		return admin_url( 'admin.php?page=jetpack#/settings?term=creator' );
+	}
+
+	/**
+	 * Get the WPCOM product slug used to make the purchase
+	 *
+	 * @return ?string
+	 */
+	public static function get_wpcom_product_slug() {
+		return 'jetpack_creator_yearly';
+	}
+
+	/**
+	 * Get the WPCOM product slug used to make the purchase
+	 *
+	 * @return ?string
+	 */
+	public static function get_wpcom_biyearly_product_slug() {
+		return 'jetpack_creator_bi_yearly';
+	}
+
+	/**
+	 * Get the WPCOM monthly product slug used to make the purchase
+	 *
+	 * @return ?string
+	 */
+	public static function get_wpcom_monthly_product_slug() {
+		return 'jetpack_creator_monthly';
+	}
+
+	/**
+	 * Checks whether the current plan (or purchases) of the site already supports the product
+	 *
+	 * @return boolean
+	 */
+	public static function has_required_plan() {
+		$purchases_data = Wpcom_Products::get_site_current_purchases();
+		if ( is_wp_error( $purchases_data ) ) {
+			return false;
+		}
+		if ( is_array( $purchases_data ) && ! empty( $purchases_data ) ) {
+			foreach ( $purchases_data as $purchase ) {
+				if ( str_starts_with( $purchase->product_slug, static::get_wpcom_biyearly_product_slug() ) ) {
+					return true;
+				}
+				if ( str_starts_with( $purchase->product_slug, static::get_wpcom_product_slug() ) ) {
+					return true;
+				}
+				if ( str_starts_with( $purchase->product_slug, static::get_wpcom_monthly_product_slug() ) ) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Checks whether the product can be upgraded - i.e. this shows the /#add-creator interstitial
+	 *
+	 * @return boolean
+	 */
+	public static function is_upgradable() {
+		$has_required_plan = self::has_required_plan();
+
+		// Mark as not upgradable if user is on unlimited tier or does not have any plan.
+		if ( ! $has_required_plan ) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Activates the product by installing and activating its plugin
+	 *
+	 * @param bool|WP_Error $current_result Is the result of the top level activation actions. You probably won't do anything if it is an WP_Error.
+	 * @return boolean|\WP_Error
+	 */
+	public static function do_product_specific_activation( $current_result ) {
+
+		$product_activation = parent::do_product_specific_activation( $current_result );
+
+		if ( is_wp_error( $product_activation ) && 'module_activation_failed' === $product_activation->get_error_code() ) {
```

Comment:
I don't think you'd end up getting the `module_activation_failed ` code here since this class only extends the base `Product` class, which does not return that error.